runtime error (panic) when flattening an included Taskfile with a default task defined

I'm excited for the new [flatten](https://github.com/go-task/task/pull/1704) feature that was [recently merged to main](https://github.com/go-task/task/commit/ef3b8537281e10e6dca2e1c8491e9ab85ea4e744) and wanted to give it a try and came across this `SIGSEGV` panic...

```
# Taskfile.yml
version: '3'
includes:
  base: 
    taskfile: Taskfile.base.yml
    flatten: true

tasks:
  foo:
    - echo "foo"
---
# Taskfile.base.yml
version: '3'
tasks:
  default: # <-- this appears to be the culprit
    - echo "bar"
```

This panics ...
```
$ task -a
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x865a21]

goroutine 6 [running]:
github.com/go-task/task/v3/taskfile/ast.(*Tasks).Merge(0xc00018c5a8, {{{0xc00011c740, 0x1, 0x1}, 0xc0001fa780}}, 0xc0001b25b0, 0xc00005e700?)
        /home/vscode/go/pkg/mod/github.com/go-task/task/v3@v3.38.1-0.20240826211905-a72b65b3b23c/taskfile/ast/tasks.go:122 +0x221
github.com/go-task/task/v3/taskfile/ast.(*Taskfile).Merge(0xc00018c500, 0xc00018cf00, 0xc0001b25b0)
        /home/vscode/go/pkg/mod/github.com/go-task/task/v3@v3.38.1-0.20240826211905-a72b65b3b23c/taskfile/ast/taskfile.go:58 +0x1eb
github.com/go-task/task/v3/taskfile/ast.(*TaskfileGraph).Merge.func1()
        /home/vscode/go/pkg/mod/github.com/go-task/task/v3@v3.38.1-0.20240826211905-a72b65b3b23c/taskfile/ast/graph.go:92 +0xdc
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /home/vscode/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x56
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        /home/vscode/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x96
```

Looks like a nil pointer at [taskfiles/ast/tasks.go:122](https://github.com/go-task/task/blob/main/taskfile/ast/tasks.go#L122)

---

If I change the included task to `bar` ...

```
# Taskfile.base.yml
version: '3'
tasks:
  bar:
    - echo "bar"
```
...everything works as expected...
```
$ task -a
task: Available tasks for this project:
* bar:       
* foo: 
```

---

- Task version: `v3.38.1-0.20240826211905-a72b65b3b23c (h1:01ZCV3qmpxFkx17tDEDO4qQ/2uzXorCUl3lDtl56nVk=)`
- Operating system: `Linux colima 6.5.0-15-generic #15-Ubuntu SMP PREEMPT_DYNAMIC Tue Jan  9 17:03:36 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`
- Experiments enabled: `none`

--- a/task_test.go
+++ b/task_test.go
@@ -1240,6 +1240,8 @@ func TestIncludesFlatten(t *testing.T) {
 		expectedOutput string
 	}{
 		{name: "included flatten", taskfile: "Taskfile.yml", task: "gen", expectedOutput: "gen from included\n"},
+		{name: "included flatten with default", taskfile: "Taskfile.yml", task: "default", expectedOutput: "default from included flatten\n"},
+		{name: "included flatten can call entrypoint tasks", taskfile: "Taskfile.yml", task: "from_entrypoint", expectedOutput: "from entrypoint\n"},
 		{name: "included flatten with deps", taskfile: "Taskfile.yml", task: "with_deps", expectedOutput: "gen from included\nwith_deps from included\n"},
 		{name: "included flatten nested", taskfile: "Taskfile.yml", task: "from_nested", expectedOutput: "from nested\n"},
 		{name: "included flatten multiple same task", taskfile: "Taskfile.multiple.yml", task: "gen", expectedErr: true, expectedOutput: "task: Found multiple tasks (gen) included by \"included\"\""},

--- a/taskfile/ast/tasks.go
+++ b/taskfile/ast/tasks.go
@@ -112,12 +112,12 @@ func (t1 *Tasks) Merge(t2 Tasks, include *Include, includedTaskfileVars *Vars) e
 		return nil
 	})
 
-	// If the included Taskfile has a default task and the parent namespace has
+	// If the included Taskfile has a default task, being not flattened and the parent namespace has
 	// no task with a matching name, we can add an alias so that the user can
 	// run the included Taskfile's default task without specifying its full
 	// name. If the parent namespace has aliases, we add another alias for each
 	// of them.
-	if t2.Get("default") != nil && t1.Get(include.Namespace) == nil {
+	if t2.Get("default") != nil && t1.Get(include.Namespace) == nil && !include.Flatten {
 		defaultTaskName := fmt.Sprintf("%s:default", include.Namespace)
 		t1.Get(defaultTaskName).Aliases = append(t1.Get(defaultTaskName).Aliases, include.Namespace)
 		t1.Get(defaultTaskName).Aliases = slices.Concat(t1.Get(defaultTaskName).Aliases, include.Aliases)

--- a/testdata/includes_flatten/Taskfile.with_default.yml
+++ b/testdata/includes_flatten/Taskfile.with_default.yml
@@ -1,0 +1,3 @@
+version: '3'
+tasks:
+  default: echo "default from included flatten"

--- a/testdata/includes_flatten/Taskfile.yml
+++ b/testdata/includes_flatten/Taskfile.yml
@@ -5,9 +5,11 @@ includes:
     taskfile: ./included
     dir: ./included
     flatten: true
+  with_default:
+    taskfile: ./Taskfile.with_default.yml
+    flatten: true
 
 tasks:
-  default:
-    cmds:
-      - echo root_directory > root_directory.txt
+  from_entrypoint: echo "from entrypoint"
+
 

